src/core/StelCore.cpp

```diff
@@ -1395,83 +1410,98 @@ double StelCore::getUTCOffset(const double JD) const
 		qWarning() << "Invalid timezone: " << tzName;
 	}
 
-	qint64 shiftInSeconds = 0;
-	if (tzName=="system_default" || (loc.planetName=="Earth" && !tzValid && !QString("LMST LTST").contains(tzName)))
+	// Complete the cache object and hash it.
+	u.loc= & const_cast<StelLocation &>(loc);
+	u.tz = & const_cast<QTimeZone&>(tz);
+	size_t dateLocHash=qHash(&u);
```

Comment:
Are you hashing a pointer to a local variable?..

Reply:
Did I misunderstand the qHash() docs? Does this only hash the pointer, not its contents? 
OK, that's not what I wanted... :-o
Do I need to write my own qHash function for the struct?

Reply:
Yes, you need a custom hash function for every custom data type. [This](https://stackoverflow.com/q/2590677/673852) may be of help.